tests/test_cairo.py

```diff
@@ -0,0 +1,131 @@
+    },
+    {
+        "inputs": [],
+        "name": "symbol",
+        "outputs": [{"name": "symbol", "type": "felt"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+    {
+        "inputs": [],
+        "name": "totalSupply",
+        "outputs": [{"name": "totalSupply", "type": "Uint256"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+    {
+        "inputs": [],
+        "name": "decimals",
+        "outputs": [{"name": "decimals", "type": "felt"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "account", "type": "felt"}],
+        "name": "balanceOf",
+        "outputs": [{"name": "balance", "type": "Uint256"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "owner", "type": "felt"}, {"name": "spender", "type": "felt"}],
+        "name": "allowance",
+        "outputs": [{"name": "remaining", "type": "Uint256"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "recipient", "type": "felt"}, {"name": "amount", "type": "Uint256"}],
+        "name": "transfer",
+        "outputs": [{"name": "success", "type": "felt"}],
+        "type": "function",
+    },
+    {
+        "inputs": [
+            {"name": "sender", "type": "felt"},
+            {"name": "recipient", "type": "felt"},
+            {"name": "amount", "type": "Uint256"},
+        ],
+        "name": "transferFrom",
+        "outputs": [{"name": "success", "type": "felt"}],
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "spender", "type": "felt"}, {"name": "amount", "type": "Uint256"}],
+        "name": "approve",
+        "outputs": [{"name": "success", "type": "felt"}],
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "spender", "type": "felt"}, {"name": "added_value", "type": "Uint256"}],
+        "name": "increaseAllowance",
+        "outputs": [{"name": "success", "type": "felt"}],
+        "type": "function",
+    },
+    {
+        "inputs": [
+            {"name": "spender", "type": "felt"},
+            {"name": "subtracted_value", "type": "Uint256"},
+        ],
+        "name": "decreaseAllowance",
+        "outputs": [{"name": "success", "type": "felt"}],
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "recipient", "type": "felt"}, {"name": "amount", "type": "Uint256"}],
+        "name": "mint",
+        "outputs": [],
+        "type": "function",
+    },
+    {
+        "inputs": [{"name": "user", "type": "felt"}, {"name": "amount", "type": "Uint256"}],
+        "name": "burn",
+        "outputs": [],
+        "type": "function",
+    },
+]
+
+
+def test_cairo_abi():
+    abi = ContractType.parse_obj({"abi": CAIRO_ERC20_ABI}).abi
+
+    # Verify struct
+    struct = abi[0]
+    assert struct.type == "struct"
+    assert struct.size == 2
```

Comment:
Can we also show that the extra fields are ending up in the ABI when it gets serialized? Otherwise, it might break for Cairo use cases

Reply:
We parse the raw data into a class, test that the values are accessible on the objects here.

Now, I just added more assertions to make sure the extra fields still show up in `data()` off the class. So the fields seem to always be there!

Does this cover all the bases?